### ISSUE TITLE
[Bug]: Bulk insert data into partition success when enable partition key

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus-sdk-go#538]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

milvus image: 2.2.0-20230801-53a23fc8
go-sdk version: branch v2.2.x

1. create a collection with an int64 partition key field with 10 num partition
2. bulk insert empty files into one partition but success. But expected return error like `not support manually specifying the partition names if partition key mode is used`

code snippet：
```
// list partitions -> success
	partitions, err := mc.ShowPartitions(ctx, collName)
// bulk insert -> error
	_, err = mc.BulkInsert(ctx, collName, partitions[0].Name, []string{""})
	common.CheckErr(t, err, false, "not support manually specifying the partition names if partition key mode is used")
```
```
2023/08/03 01:13:12 milvus_client.go:13: (ApiRequest): func [BulkInsert], args: [context.Background.WithDeadline(2023-08-03 01:14:12.24385 +0800 CST m=+60.004955459 [59.322139291s]) CjIFWr _default_0 [] []]
2023/08/03 01:13:12 milvus_client.go:20: (ApiResponse): func [BulkInsert], results: [443255264145552354]
```

### Expected Behavior

error

### Steps To Reproduce

```markdown
partition_key_test.go
 case name: TestPartitionKeyPartitionOperation
```

**Comment by yanliang567** [comment]
@ThreadDao I think it would be fixed by [milvus/io:#26129](https://github.com/milvus-io/milvus/pull/26129)

**Comment by ThreadDao** [comment]
fixed
